text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,542 @@
+# Context Propagation: A Layered Approach 
+
+* [Motivation](#Motivation)
+* [OpenTelemetry layered architecture](#OpenTelemetry-layered-architecture)
+  * [Cross-Cutting Concerns](#Cross-Cutting-Concerns)
+    * [Observability API](#Observability-API)
+    * [Baggage API](#Baggage-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+  * [Global initialization](#Global-initialization)
+  * [Extracting and injecting from HTTP headers](#Extracting-and-injecting-from-HTTP-headers)
+  * [Simplfy the API with automated context propagation](#Simplfy-the-API-with-automated-context-propagation)
+  * [Implementing a propagator](#Implementing-a-propagator)
+  * [Implementing a concern](#Implementing-a-concern)
+  * [The scope of current context](#The-scope-of-current-context)
+  * [Referencing multiple contexts](#Referencing-multiple-contexts)
+  * [Falling back to explicit contexts](#Falling-back-to-explicit-contexts)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+# Motivation
+
+This RFC addresses the following topics:
+
+**Separatation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Baggage) and observability data (SpanContext, etc) 
+  separately.  
+
+**Extensibility**
+* Allow developers to create new applications for context propagation. For 
+  example: A/B testing, encrypted or authenticated data, and new, experimental 
+  forms of observability.
+
+## Explanation
+
+# OpenTelemetry layered architecture
+
+The design of OpenTelemetry is based on the priciples of aspect-oriented 
+programming, adopted to the needs of distributed systems.
+
+OpenTelemetry is separated into two layers: **cross-cutting concerns** which 
+intertwine with a program's application logic, and cannot be encapsulated. In 
+this architecture, each concern is modeled as an independent subsystem. 
+Multiple concerns - including the tracing and baggage systems provided 
+by OpenTelemetry - then share the same underlying **context propagation** 
+system, which allows these cross-cutting concerns to store and access their 
+data across the lifespan of a distributed transaction.
+
+# Cross-Cutting Concerns
+
+## Observability API
+Distributed tracing is one example of a cross-cutting concern. Tracing code is 
```

Comment:
can we add an example here?

Reply:
The above section flushes this out with a comparison to logging. Is there another kind of example you would like to see?